### DISCUSSION TITLE
Cypress erroring on ng-dialog [cdk-overlay-container]

### DISCUSSION THREAD

**Discussion by andersonterry** [cypress-io/cypress#17947] [Questions and Help]
note:  cross posted to stackOverflow here:  [https://stackoverflow.com/questions/68990991/cypress-will-not-display-ng-dialog-cdk-overlay-container]

**Markup**

<button data-testid="usertable.editbutton"...

**Description**

Cypress issues a client script error (stack below) after this btn.click(), which generates `<div class="cdk-overlay-container">` when clicked manually. The app and dependent services are hosted in Azure, if that matters.

Any ideas appreciated. My expectation is:

```
btn.click().then(()=>{ 
 // cdk-overlay-container available here
})
```

**Spec**

```
describe('sample', () => {
    beforeEach(() => {
      cy.fixture('example').as('fixture') 
    })
    it('login', () => {
        cy.fixture('example').then((f) => {
            cy.visit(f.baseUrl).wait(f.pageWaitDefault);
        })
    })
    it('click table row edit button', () => {
          cy.contains('An Edited Test User 1').parent('tr').within(()=>{
            cy.get('[data-testid="usertable.editbutton"]').click() //errors here
        })
    })
})
```

**Client Error**

TypeError: Cannot read property 'length' of null at https://pic-authui-dev.azurewebsites.net/vendor.js:24819:32 at Array.forEach () at HttpHeaders.lazyInit (https://pic-authui-dev.azurewebsites.net/vendor.js:24813:38) at HttpHeaders.init (https://pic-authui-dev.azurewebsites.net/vendor.js:24917:22) at HttpHeaders.forEach (https://pic-authui-dev.azurewebsites.net/vendor.js:24984:14) at Observable._subscribe (https://pic-authui-dev.azurewebsites.net/vendor.js:26426:25) at Observable._trySubscribe (https://pic-authui-dev.azurewebsites.net/vendor.js:128303:25) at Observable.subscribe (https://pic-authui-dev.azurewebsites.net/vendor.js:128289:22) at innerSubscribe (https://pic-authui-dev.azurewebsites.net/vendor.js:129306:23) at MergeMapSubscriber._innerSub (https://pic-authui-dev.azurewebsites.net/vendor.js:130758:98)

**Comment by portyu9** [comment]
Hi @andersonterry 

The `cdk-overlay-container` is not the problem. Your stack trace shows Angular failing while building `HttpHeaders`, most likely because an auth/header value is `null`.

Your tests are also dependent on each other:

```js
it('login', ...)
it('click table row edit button', ...)
```

Cypress clears cookies/localStorage between tests, so the second test should not rely on authentication established by the first one.

Move the login/setup into `beforeEach()` (or use `cy.session()`):

```js
beforeEach(() => {
  cy.login()
  cy.visit('/')
})

it('opens the edit dialog', () => {
  cy.contains('An Edited Test User 1')
    .parents('tr')
    .find('[data-testid="usertable.editbutton"]')
    .click()

  cy.get('.cdk-overlay-container')
    .should('be.visible')
})
```

Also fix the Angular interceptor so it never adds a `null` header value.

So the fix is: **restore authentication for each test and do not pass `null` into Angular `HttpHeaders`**. Once the request succeeds, the Material dialog should render normally.

Please mark this answer as accepted if it helped.
